src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -313,7 +340,7 @@ class BookingExtractionService(
     val mostRecentReleaseIsPrrd = mostRecentSentencesByReleaseDate.any { it.releaseDateTypes.getReleaseDateTypes().contains(PRRD) }
     // For now we can't calculate HDCED if there is a consecutive sentence with EDS or SOPC sentences
     if (!mostRecentReleaseIsPrrd && sentences.none { it is ConsecutiveSentence && it.hasAnyEdsOrSopcSentence() }) {
-      val latestNonRecallRelease = extractionService.mostRecentSentenceOrNull(sentences.filter { !it.isRecall() }, SentenceCalculation::releaseDate)
+      val latestNonRecallRelease = extractionService.mostRecentSentenceOrNull(sentences.filter { !it.isRecall() && it !is DetentionAndTrainingOrderSentence }, SentenceCalculation::releaseDate)
```

Comment:
also single term? Also consecutive dto? Again I think we need this:

`it is DetentionAndTrainingOrderSentence || (it is ConsecutiveSentence && it.orderedSentences.all{ it is DetentionAndTrainingOrderSentence }) || it is DtoSingleTermSentence`

Reply:
Done.